internal/pkg/flink/internal/controller/table_controller.go

```diff
@@ -191,6 +197,7 @@ func (t *TableController) refreshResults(ctx context.Context, statement types.Pr
 		for {
 			select {
 			case <-ctx.Done():
+				go t.store.DeleteStatement(statement.StatementName)
```

Comment:
Was it necessary to move the delete statement here?

Reply:
I did this to make sure we are still deleting statements that are in RUNNING state. We don't want to delete COMPLETED statements anymore, but I think we should still delete the ones that are running. What do you think?

Reply:
Oh yeah absolutely, good catch!